lib/keystore/helpers.go

```diff
@@ -148,6 +148,22 @@ func LoadKeystore(key string, keyStore TyperInserter, keyRing KeyRing) (err erro
 		return keyStore.Insert(keyRing.Heather())
 	case "ian":
 		return keyStore.Insert(keyRing.Ian())
+	case "default":
+		if keyStore.Type() == "ed25519" {
+			kp, err := ed25519.GenerateKeypair()
+			if err != nil {
+				return err
+			}
+
+			return keyStore.Insert(kp)
+		} else {
+			kp, err := sr25519.GenerateKeypair()
+			if err != nil {
+				return err
+			}
+
+			return keyStore.Insert(kp)
+		}
```

Comment:
We currently only have the test key flow in the CLI. Check https://github.com/1garo/gossamer/blob/feat/generate-account-if-not-specified/cmd/gossamer/commands/root.go#L577

I feel we should separate out the flow for the keystore setup in the `cmd` package instead of this change. The rust implementation has `--dev` flag; maybe we could add the same!

Reply:
you mean something like [subkey?](https://docs.substrate.io/reference/command-line-tools/subkey/)

Reply:
Maybe `account generate` already do that, I don't know if I really understood what you meant

Reply:
I spent some more time going through the code. Thank you for your patience and time!

I believe we don't have the need to generate an account if not specified since we're still in development and only need development accounts.

If we want to support the use of auto-generated or external accounts, we'll have to
- Add a new flag `dev` to enable the clear separation for the development runs
- Clear separation for the development config
- Store the auto generated account in disk
- Ability to load the account from disk

I think this isn't a priority for us at the moment. I'll update the issue.